UndertaleModTool/Converters/UndertaleCachedImageLoader.cs

```diff
@@ -158,42 +150,47 @@ public static void Reset()
             currBufferSize = 1048576;
         }
 
-        public static Bitmap CreateSpriteBitmap(Rectangle rect, in UndertaleTexturePageItem texture, int diffW = 0, int diffH = 0, bool isTile = false)
+        public static SKBitmap CreateSpriteBitmap(Rect rect, in UndertaleTexturePageItem texture, int diffW = 0, int diffH = 0, bool isTile = false)
         {
-            using MemoryStream stream = new(texture.TexturePage.TextureData.TextureBlob);
-            Bitmap spriteBMP = new(rect.Width, rect.Height);
+            SKBitmap spriteBMP = new((int)rect.Width, (int)rect.Height);
 
             rect.Width -= (diffW > 0) ? diffW : 0;
             rect.Height -= (diffH > 0) ? diffH : 0;
             int x = isTile ? texture.TargetX : 0;
             int y = isTile ? texture.TargetY : 0;
 
-            using (Graphics g = Graphics.FromImage(spriteBMP))
+            using (SKCanvas g = new(spriteBMP))
             {
-                using Image img = Image.FromStream(stream); // "ImageConverter.ConvertFrom()" does the same, except it doesn't explicitly dispose MemoryStream
-                g.DrawImage(img, new Rectangle(x, y, rect.Width, rect.Height), rect, GraphicsUnit.Pixel);
+                using SKBitmap img = SKBitmap.Decode(texture.TexturePage.TextureData.TextureBlob);
+                
+                var rectDest = SKRect.Create(x, y, (float)rect.Width, (float)rect.Height);
+                var rectSrc = SKRect.Create((float)rect.Left, (float)rect.Top, (float)rect.Width, (float)rect.Height);
+                g.DrawBitmap(img, rectSrc, rectDest);
             }
+            spriteBMP.SetImmutable();
```

Comment:
why do you need to set it immutable?

Reply:
I think it should have a better performance like this, though I'm not sure about this particular one.

Reply:
But the returned bitmap shouldn't be changed ever since, so I think this will not hurt

Reply:
If I read the docs correctly, it doesnt give a performance boost and is just meant as a "mark this image as read only and thus crash if someone tries to write to its pixels.". It thus seemed relatively pointless to me, unless we actually do want that.

Reply:
I double checked, it doesnt even crash. it just does nothing at all.

Reply:
So should I remove it or not?

Reply:
unless you have a very good reason for them to stay, I would prefer it if you'd remove them, yeah.

Reply:
Only this one, the second one actually _should_ give a boost (at least theoretically).